### PR TITLE
[1LP][WIPTEST] smartstate smoketest in pod appliance

### PR DESCRIPTION
**Pull request by izapolsk**
podified appliance buids are shipped with smartstate image for aws. This image has to be smoke tested.
This PR should introduce smoke tests for aws image 

REQUIRES: https://github.com/ManageIQ/wrapanapi/pull/358

{{ pytest: --long-running --use-template-cache --use-provider=cm-pod2 cfme/tests/pod/test_appliance_crud.py::test_aws_smartstate_pod }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 362bf6edc16b824221b8c68286a3466e89f10fcc:

`cfme/tests/pod/test_appliance_crud.py`:
- :bangbang: Line 195:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 547:21: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3082446bdfc2b37cb45f634bacc28437293da100:

`cfme/tests/pod/test_appliance_crud.py`:
- :bangbang: Line 192:5: **E303** *too many blank lines (2)*
- :bangbang: Line 204:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 556:21: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3cc8fd59f49c8cdf2256f829d4ddfa546d05c963:

`cfme/tests/pod/test_appliance_crud.py`:
- :bangbang: Line 195:5: **E303** *too many blank lines (2)*
- :bangbang: Line 207:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 559:21: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f0bb301f85a9e6c9f3a1bd4f1ce77a17a4d2c7bd:

`cfme/tests/pod/test_appliance_crud.py`:
- :bangbang: Line 195:5: **E303** *too many blank lines (2)*
- :bangbang: Line 210:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 562:21: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f3dcf67f1a90ebb6eebfb703f44e62df9e5b72cd:

`cfme/tests/pod/test_appliance_crud.py`:
- :bangbang: Line 195:5: **E303** *too many blank lines (2)*
- :bangbang: Line 210:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 562:21: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f69de81e913a9614202757d3f9f87de7038122ae:

`cfme/tests/pod/test_appliance_crud.py`:
- :bangbang: Line 195:5: **E303** *too many blank lines (2)*
- :bangbang: Line 210:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 562:21: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 86b88bf4ff526f6419d721082f6ead8728cdbdbe:

`cfme/tests/pod/test_appliance_crud.py`:
- :bangbang: Line 195:5: **E303** *too many blank lines (2)*
- :bangbang: Line 210:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 562:21: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1dd5b5b22070adad8820a0baf1d79ab23156618a:

`cfme/tests/pod/test_appliance_crud.py`:
- :bangbang: Line 203:5: **E303** *too many blank lines (2)*
- :bangbang: Line 569:21: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e83abfbad3c3f92cd43ec63e1143263bef21db89:

`cfme/tests/pod/test_appliance_crud.py`:
- :red_circle: Line 1:1: **F401** *'copy' imported but unused*
- :red_circle: Line 192:5: **F841** *local variable 'ss_image_name' is assigned to but never used*
- :bangbang: Line 203:5: **E303** *too many blank lines (2)*
- :bangbang: Line 544:1: **E302** *expected 2 blank lines, found 0*
- :red_circle: Line 563:5: **F841** *local variable 'backup_settings' is assigned to but never used*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/pod/test_appliance_crud.py [hunk 1]
```diff
@@ -492,8 +588,36 @@ def test_aws_smartstate_pod():
         casecomponent: Containers
         caseimportance: medium
         initialEstimate: 1h
+        testSteps:
```
Comment (mshriver) [inline cfme/tests/pod/test_appliance_crud.py]: This should be numbered - but isn't necessary for an automated test.  Please either remove, or number the list.

**Comment by izapolsk** [comment]
as I see in PRT I'll need to install and setup docker in vm.

File: cfme/tests/pod/test_appliance_crud.py [hunk 2]
```diff
@@ -492,8 +588,36 @@ def test_aws_smartstate_pod():
...
+          1. pull smartstate image from registry
+          2. tag it accordingly and push to externally available registry
+          3. setup appliance to use that image for smartstate in aws
+          4. add aws provider
+          5. find 24/7 vm in aws and perform smartstate analysis
     """
-    pass
+    appliance = temp_ssa_pod_appliance
+
+    if BZ(1684203, forced_streams=['5.9', '5.10']).blocks:
```
Comment (jawatts) [inline cfme/tests/pod/test_appliance_crud.py]: We can remove 5.9 now since breakpoint has been set.

File: cfme/tests/pod/test_appliance_crud.py [hunk 3]
```diff
@@ -325,6 +330,98 @@ def ipa_user(temp_pod_appliance, ipa_auth_provider):
     appliance.server.login_admin()
 
 
+@pytest.fixture
+def aws_provider(temp_pod_appliance):
+    from cfme.utils.providers import get_crud
+    prov = get_crud('ec2west')
```
Comment (jawatts) [inline cfme/tests/pod/test_appliance_crud.py]: **Question** Is there a reason this is hardcoded in? If the name changes, the test will break.

File: cfme/tests/pod/test_appliance_crud.py [hunk 4]
```diff
@@ -325,6 +330,98 @@ def ipa_user(temp_pod_appliance, ipa_auth_provider):
...
+    ssa_settings = {'ems': {
+        'ems_amazon': {
+            'agent_coordinator': {
+                'docker_image': '{stream}:{version}'.format(stream=SSA_IMAGE_STREAM,
+                                                            version=ssa_data['version']),
+                'docker_registry': ssa_data['registry'],
+            }
+        }
+    }}
+    if BZ(1684203, forced_streams=['5.9', '5.10']).blocks:
```
Comment (jawatts) [inline cfme/tests/pod/test_appliance_crud.py]: with 5.9 breakpoint, I think we can remove 5.9 here.

**Comment by jawatts** [comment]
Moving to WIPTEST as docker needs to be setup in vm to pass PRT

**Pull request merged**